make_count/make_count.tex

```diff
@@ -0,0 +1,293 @@
+\newcommand{\MultiSets}{\mathrm{MultiSets}}
+\newcommand{\len}{\mathrm{len}}
+\newcommand{\din}{\mathrm{d_{in}}}
+\newcommand{\dout}{\mathrm{d_{out}}}
+\newcommand{\Relation}{\mathrm{Relation}}
+\newcommand{\question}[1]{\textcolor{red}{\textbf{Question:} #1}}
+\newcommand{\ellOne}{\mathrm{\ell_1}}
+\newcommand{\maxUsize}{\texttt{usize::MAX}}
+\newcommand{\function}{\texttt{function}}
+\newcommand{\True}{\texttt{True}}
+\newcommand{\todo}[1]{{\begin{center} \textcolor{teal}{{\huge TODO:} #1}
+\end{center}}}
+\newcommand{\mike}[1]{{ {\color{green}{(mike)~#1}}}}
+
+% convert ddmmyyyy to dd MONTH year
+\def\changeformat#1{\xchangeformat#1\relax}
+\def\xchangeformat#1/#2/#3\relax{%
+#1 %
+\ifcase#2 \or
+January\or February\or March\or April\or May\or June\or July\or
+August\or September\or October\or November\or December\fi
+\ #3}
+
+\newcommand{\permalink}[2]{This proof is based on the code in \url{#1} from \changeformat{#2}.}
+
+\setlength{\parindent}{0.0in}
+\setlength{\parskip}{0.05in}
+\title{Privacy Proofs for OpenDP: MakeCount}
+\author{Connor Wagaman -- wagaman@college.harvard.edu}
+
+\usepackage{xcolor}
+\usepackage{amsthm}
+\newtheorem{theorem}{Theorem}[section]
+\newtheorem{lemma}[theorem]{Lemma}
+
+\theoremstyle{definition}
+\newtheorem{remark}{Remark}
+\newtheorem{definition}{Definition}[section]
+\newtheorem{observation}{Observation}
+\newtheorem{note}{Note}
+\newtheorem{hope}{Hope}
+\newtheorem{warning}{Warning}
+\newtheorem{problem}{Problem}
+\newtheorem{fear}{Fear}
+
+\begin{document}
+
+\maketitle
+
+\tableofcontents
+
+\section{MakeCount}
+
+\subsection{Implementation of MakeCount in Rust}
+\label{sub:rust-implementation}
+
+\inOpenDPRust{make\_count}{https://github.com/opendp/opendp/blob/main/rust/opendp/src/trans/count.rs}
+
+\permalink{https://github.com/opendp/opendp/blob/c3b5c3bd9fc50c556362b628f08c5fddea069b4d/rust/opendp/src/trans/count.rs\#L14-L27}{12/07/2021} (It is from \href{https://github.com/opendp/opendp/pull/173}{this pull request}.)
+The Rust code can also be seen below. 
+
+\begin{lstlisting}[language = Rust]
+pub fn make_count<TIA, TO>(
+) -> Fallible<Transformation<VectorDomain<AllDomain<TIA>>, AllDomain<TO>, SymmetricDistance, AbsoluteDistance<TO>>>
+    
+    where TO: ExactIntCast<usize> + One + DistanceConstant<IntDistance>, IntDistance: InfCast<TO> {
+    
+    Ok(Transformation::new(
+        VectorDomain::new_all(),
+        AllDomain::new(),
+        // think of this as: min(arg.len(), TO::MAX_CONSECUTIVE)
+        Function::new(move |arg: &Vec<TIA>|
+            TO::exact_int_cast(arg.len()).unwrap_or(TO::MAX_CONSECUTIVE)),
+        SymmetricDistance::default(),
+        AbsoluteDistance::default(),
+        StabilityRelation::new_from_constant(TO::one())))
+}
+\end{lstlisting}
+
+\subsection{Implementation of MakeCount in Python-style pseudocode, with preconditions}
+\label{sec:python-pseudocode}
+
+We now use Python-style pseudocode to present a representation of the Rust function.

```

Comment:
I like this header text, though perhaps it should appear somewhere in the documentation site? @pdurbin